eclipse-theia/theia
# Search in Workspace: fix history for programmatically added text

#### What it does

Fixes https://github.com/eclipse-theia/theia/issues/10437 .

#### How to test


1. Open Theia to e.g. the Theia repo. Open the search in workspace widget. Type a few things into the search field, pausing in between to let the search & history update happen. Use the up-and-down arrows to view the contents of the history.
2. Open an editor, select an interesting word in it, and hit ctrl+shift+F to search for it. It will appear in the search input field of the SIW widget. Now use the up-and-down arrows to view the contents of the history, and the programmatically-added search string will be found in the history as it should be.


#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
